### PR TITLE
Use quarkus-test-profiles

### LINKED ISSUE #217: Add integration tests that start the quarkus server with each backing store
While testing, I realized that the dynamo backing store initialization within quarkus was broken. We need to get tests integration tests in place that actually test the stores in the server, not just independently.

### PR DESCRIPTION
**Pull request by snazy**
Prepare `:nessie-quarkus` to be able to test different version-stores/database-types.

More test-profiles will be added when TVS2 finally lands.

Fixes #217

---
This change is [Reviewable](https://reviewable.io/reviews/projectnessie/nessie/1924)

### CODE REVIEW

File: servers/quarkus-server/src/test/java/org/projectnessie/server/profiles/QuarkusNativeProfileInmemory.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+public class QuarkusNativeProfileInmemory implements QuarkusTestProfile {
+
+  @Override
+  public Map<String, String> getConfigOverrides() {
+    return ImmutableMap.of("nessie.version.store.type", "INMEMORY");
+  }
+
+  @Override
+  public String getConfigProfile() {
+    return "prod";
```
Comment (dimas-b) [inline servers/quarkus-server/src/test/java/org/projectnessie/server/profiles/QuarkusNativeProfileInmemory.java:31]: How does this work with unit test that generally use the `test` quarkus profile, if I'm not mistaken?

File: servers/quarkus-server/src/test/java/org/projectnessie/server/TestBasicOperations.java [hunk 2]
```diff
@@ -16,6 +16,9 @@
 package org.projectnessie.server;
 
 import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.TestProfile;
+import org.projectnessie.server.profiles.QuarkusNativeProfileInmemory;
 
 @QuarkusTest
+@TestProfile(QuarkusNativeProfileInmemory.class)
```
Comment (dimas-b) [inline servers/quarkus-server/src/test/java/org/projectnessie/server/TestBasicOperations.java:23]: Do we need to apply this to _unit_ tests that run Quarkus in a JVM?

File: servers/quarkus-server/src/test/java/org/projectnessie/server/profiles/QuarkusNativeProfileInmemory.java (hunk 1 above)
Comment (snazy) [reply servers/quarkus-server/src/test/java/org/projectnessie/server/profiles/QuarkusNativeProfileInmemory.java:31]: native-tests require the `prod`, not the `test` profile, sadly

File: servers/quarkus-server/src/test/java/org/projectnessie/server/TestBasicOperations.java (hunk 2 above)
Comment (snazy) [reply servers/quarkus-server/src/test/java/org/projectnessie/server/TestBasicOperations.java:23]: argh - damn copy+paste bug. should be `QuarkiusProfileInmemory`

Comment (snazy) [reply servers/quarkus-server/src/test/java/org/projectnessie/server/TestBasicOperations.java]: good catch BTW!

File: servers/quarkus-server/src/test/java/org/projectnessie/server/profiles/QuarkusNativeProfileInmemory.java (hunk 1 above)
Comment (dimas-b) [reply servers/quarkus-server/src/test/java/org/projectnessie/server/profiles/QuarkusNativeProfileInmemory.java:31]: yeah, I meant using this in `TestBasicOperations`, which is fixed now :+1:

**Review by dimas-b — APPROVED** [review]

**Pull request merged**